lib/routes/charge/items/item_page.dart

```diff
@@ -0,0 +1,275 @@
+import 'package:breez/bloc/account/account_actions.dart';
+import 'package:breez/bloc/account/account_bloc.dart';
+import 'package:breez/bloc/account/account_model.dart';
+import 'package:breez/bloc/account/fiat_conversion.dart';
+import 'package:breez/bloc/blocs_provider.dart';
+import 'package:breez/bloc/pos_catalog/actions.dart';
+import 'package:breez/bloc/pos_catalog/bloc.dart';
+import 'package:breez/bloc/pos_catalog/model.dart';
+import 'package:breez/bloc/user_profile/currency.dart';
+import 'package:breez/theme_data.dart' as theme;
+import 'package:breez/widgets/back_button.dart' as backBtn;
+import 'package:breez/widgets/flushbar.dart';
+import 'package:breez/widgets/single_button_bottom_bar.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+import '../currency_wrapper.dart';
+
+class ItemPage extends StatefulWidget {
+  final Item item;
+  final PosCatalogBloc _posCatalogBloc;
+
+  ItemPage(this._posCatalogBloc, {this.item});
+
+  @override
+  State<StatefulWidget> createState() {
+    return ItemPageState();
+  }
+}
+
+class ItemPageState extends State<ItemPage> {
+  final GlobalKey<ScaffoldState> _scaffoldKey = GlobalKey<ScaffoldState>();
+
+  final _formKey = GlobalKey<FormState>();
+  TextEditingController _nameController = TextEditingController();
+  TextEditingController _priceController = TextEditingController();
+  TextEditingController _skuController = TextEditingController();
+  AccountBloc _accountBloc;
+  bool _isInit = false;
+  String _title = "Add Item";
+  CurrencyWrapper _selectedCurrency = CurrencyWrapper.fromBTC(Currency.BTC);
+
+  @override
+  void didChangeDependencies() {
+    if (!_isInit) {
+      _accountBloc = AppBlocsProvider.of<AccountBloc>(context);
+      FetchRates fetchRatesAction = FetchRates();
+      _accountBloc.userActionsSink.add(fetchRatesAction);
+      if (widget.item != null) {
+        _accountBloc.accountStream.first.then((accountModel) {
+          setState(() {
+            _title = "Edit Item";
+            _nameController.text = widget.item.name;
+            _skuController.text = widget.item.sku;
+            _selectedCurrency = CurrencyWrapper.fromShortName(
+                widget.item.currency, accountModel);
+            _priceController.text = _formattedPrice(widget.item.price);
+          });
+        });
+      }
+
+      fetchRatesAction.future.catchError((err) {
+        if (this.mounted) {
+          setState(() {
+            showFlushbar(context,
+                message: "Failed to retrieve BTC exchange rate.");
```

Comment:
Are we OK to continue with this form in a state without the conversion rates.

Reply:
I allowed the users to continue without a fiat currency list so that they can at least create items with BTC/SAT.